### PR TITLE
Minify json

### PR DESCRIPTION
**Pull request by BlackBooth**
As suggested in issue #71, added the possibility to minify the json output.

### CODE REVIEW

File: JsonFlatFileDataStore/DataStore.cs [hunk 1]
```diff
@@ -474,7 +474,7 @@ private string ReadJsonFromFile(string path)
                 catch (FileNotFoundException)
                 {
                     File.WriteAllText(path, _encryptJson(json));
-                    break;
```
Comment (ttu) [inline JsonFlatFileDataStore/DataStore.cs:477]: This is now  part of another PR and can be removed from this PR

**Review by ttu — CHANGES_REQUESTED** [review]
Thanks! Can you add a test for this. I added a test for non minified version
https://github.com/ttu/json-flatfile-datastore/blob/37a2c0a795257b0600ee8d0362cb0d4e264e1a27/JsonFlatFileDataStore.Test/DataStoreTests.cs#L478

This way we can validate that this also works when we have `useLowerCamelCase` as false.

**Comment by BlackBooth** [comment]
Merged from master, added test and fixed a bug.

File: JsonFlatFileDataStore.Test/DataStoreTests.cs [hunk 2]
```diff
@@ -493,6 +494,40 @@ public async Task FileContent_DefaultFormat(bool useLowerCamelCase)
 
             Assert.Contains(allowedLengths, i => i == content.Length);
             
+
+            UTHelpers.Down(path);
+        }
+
+        [Theory]
+        [InlineData(true, true)]
```
Comment (ttu) [inline JsonFlatFileDataStore.Test/DataStoreTests.cs:502]: We can add allowedLengths as a parameter, then no need to separate logic in the test

```cs
[InlineData(true, true, new [] { 40 } )]
```

File: JsonFlatFileDataStore.Test/DataStoreTests.cs [hunk 3]
```diff
@@ -493,6 +494,40 @@ public async Task FileContent_DefaultFormat(bool useLowerCamelCase)
...
+            var store = new DataStore(path, useLowerCamelCase: useLowerCamelCase, minifyJson: useMinifiedJson);
+            var collection = store.GetCollection<Movie>("movie");
+            await collection.InsertOneAsync(new Movie { Name = "Test", Rating = 5 });
+
+            var content = UTHelpers.GetFileContent(path);
+
+            // NOTE: File format is different depending on used OS. Windows uses \r\n and Linux \r
+            //   - "{\r\n  \"movie\": [\r\n    {\r\n      \"name\": \"Test\",\r\n      \"rating\": 5.0\r\n    }\r\n  ]\r\n}",
+            //   - "{\r  \"movie\": [\r    {\r      \"name\": \"Test\",\r      \"rating\": 5.0\r    }\r  ]\r}"
+            // Length on Windows is 81 and on Linux 74
```
Comment (ttu) [inline JsonFlatFileDataStore.Test/DataStoreTests.cs:519]: You can add to comments a minified example

File: JsonFlatFileDataStore/DataStore.cs [hunk 4]
```diff
@@ -44,9 +44,9 @@ public DataStore(string path, bool useLowerCamelCase = true, string keyProperty
                             // Serializing JObject ignores SerializerSettings, so we have to first deserialize to ExpandoObject and then serialize
                             // http://json.codeplex.com/workitem/23853
                             var jObject = JsonConvert.DeserializeObject<ExpandoObject>(data.ToString());
-                            return JsonConvert.SerializeObject(jObject, Formatting.Indented, _serializerSettings);
+                            return JsonConvert.SerializeObject(jObject, minifyJson ? Formatting.None: Formatting.Indented, _serializerSettings);
```
Comment (ttu) [inline JsonFlatFileDataStore/DataStore.cs:47]: Selected formatting could be done before setting `_toJsonFunc`.

```
var usedFormatting = minifyJson ? Formatting.None: Formatting.Indented;
```

Then there is no need to have conditional check every time function is executed.

**Review by ttu — CHANGES_REQUESTED** [review]
Looks good! Few minor improvements we could make.

Besides those we should also update the documentation. You can make the changes in this PR or I can make updates those later.
* Add this change to CHANGELOD: https://github.com/ttu/json-flatfile-datastore/blob/master/CHANGELOG.md
* Update README: https://github.com/ttu/json-flatfile-datastore#writing-to-file

**Comment by BlackBooth** [comment]
I've added the suggested changes.

One question came to my mind: Should encrypted output always be minified, since nobody can read the output anyways?

**Comment by ttu** [comment]
>  Should encrypted output always be minified, since nobody can read the output anyways?

**Pull request merged**